xds/internal/xdsclient/controller/controller.go

```diff
@@ -89,7 +89,7 @@ type Controller struct {
 }
 
 // New creates a new controller.
-func New(config *bootstrap.ServerConfig, updateHandler pubsub.UpdateHandler, validator xdsresource.UpdateValidatorFunc, logger *grpclog.PrefixLogger) (_ *Controller, retErr error) {
+func New(config *bootstrap.ServerConfig, updateHandler pubsub.UpdateHandler, validator xdsresource.UpdateValidatorFunc, logger *grpclog.PrefixLogger, opts ...grpc.DialOption) (_ *Controller, retErr error) {
```

Comment:
What about making the `grpc.Dial` call itself (below) overridable instead of the dial options?

Reply:
You mean make the user set a custom grpc.Dial function instead of passing in extra dial options? 

I am ok with either of these options, but I thought we wanted to restrict the scope of the supported functionality as part of this change. 

By allowing overriding of grpc.Dial the user for example will also be able to ignore the default options, change the server URI etc. I don't think this is an issue, but any concerns against this?

Reply:
I thought the way we were limiting it was by leaving it in `internal` and making sure only whitelisted packages could use it.  If you think we'll only ever need to add dial options to make this work, then this is fine.  Otherwise, it would be more flexible to override all of `grpc.Dial`.